scheduler/pkg/kafka/dataflow/server.go

```diff
@@ -420,7 +434,7 @@ func (c *ChainerServer) handlePipelineEvent(event coordinator.PipelineEventMsg)
 			if err != nil {
 				logger.WithError(err).Errorf("Failed to set pipeline state to terminating for %s", pv.String())
 			}
-			msg := c.createPipelineMessage(pv)
+			msg := c.createPipelineMessage(pv) // note pv is a copy and does not include the new change to terminating state
```

Comment:
This is not an issue right? I see it could lead to duplicate logs with status = pipeline.PipelineTerminated, but not sure of other implications

Reply:
It is not an issue in this particular case as we send the event (`PipelineTerminate`) to dataflow-engine and mark it in the scheduler to be `PipelineTerminating` until dataflow responds back with success or failure.